tests/extras/datasets/pandas/test_avro_dataset.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2020 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+# or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import PurePosixPath
+
+import numpy
+import pandas
+import pytest
+from fsspec.implementations.http import HTTPFileSystem
+from fsspec.implementations.local import LocalFileSystem
+from gcsfs import GCSFileSystem
+from s3fs.core import S3FileSystem
+
+from kedro.extras.datasets.pandas import AVRODataSet
+from kedro.extras.datasets.pandas.avro_dataset import infer_pandas_dtype_to_avro_type
+from kedro.io import DataSetError
+from kedro.io.core import PROTOCOL_DELIMITER
+
+FILENAME = "test.avro"
+
+
+@pytest.fixture
+@pytest.mark.parametrize("tmp_path", ["/tmp"], indirect=True)
+def filepath_avro(tmp_path):
+    return (tmp_path / FILENAME).as_posix()
+
+
+@pytest.fixture
+def dummy_data():
+    return pandas.DataFrame({"col1": [1, 2], "col2": [4, 5], "col3": [5, 6]})
+
+
+@pytest.fixture
+def avro_pd_data_set(filepath_avro):
+    return AVRODataSet(filepath=filepath_avro)
+
+
+class TestPandasAvroDataSet:
+    def test_credentials_propagated(self, mocker):
+        """Test propagating credentials for connecting to GCS"""
+        mock_fs = mocker.patch("fsspec.filesystem")
+        credentials = {"key": "value"}
+
+        AVRODataSet(filepath=FILENAME, credentials=credentials)
+
+        mock_fs.assert_called_once_with("file", **credentials)
+
+    def test_save_and_load(self, tmp_path, dummy_data):
```

Comment:
Can you divide this up into two tests, one for the happy path and one for the unhappy path?